self-hosted compiler does not map 'std' into every package

#### test.zig

```zig
const bar = @import("bar");
pub fn main() void {
    _ = bar;
}
```

#### bar.zig

```zig
const std = @import("std");

comptime {
    _ = std;
}
```

#### Actual Output

```
$ ./zig-out/bin/zig build-exe test.zig --pkg-begin bar bar.zig --pkg-end
bar.zig:1:21: error: unable to open 'std': PackageNotFound
const std = @import("std");
                    ^
```

#### Expected Output

It should succeed the build. 'builtin' and 'root' should additionally be mapped into every package.

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -1227,6 +1227,12 @@ pub fn create(gpa: *Allocator, options: InitOptions) !*Compilation {
             } else main_pkg;
             errdefer if (options.is_test) root_pkg.destroy(gpa);
 
+            var other_pkg_iter = main_pkg.table.valueIterator();
+            while (other_pkg_iter.next()) |pkg| {
+                try pkg.*.add(gpa, "builtin", builtin_pkg);
+                try pkg.*.add(gpa, "std", std_pkg);
+            }
+
             try main_pkg.addAndAdopt(gpa, "builtin", builtin_pkg);
             try main_pkg.add(gpa, "root", root_pkg);
             try main_pkg.addAndAdopt(gpa, "std", std_pkg);

